### PR TITLE
Add Bulk Action to Scheduled Reports

### PR DESCRIPTION
**Pull request by minhaminha**
## Product Description

This PR adds a bulk delete and bulk send feature to the scheduled reports page. It also adds pagination for both the main scheduled reports and others' scheduled reports list. 

Current page:
image

Updated page: 
Screen Shot 2021-10-01 at 12 14 34 PM

Bulk deletion buttons:
Screen Shot 2021-10-01 at 12 15 45 PM

Confirmation windows: 
Screen Shot 2021-10-01 at 12 17 45 PM

Screen Shot 2021-10-01 at 12 18 07 PM

## Technical Summary

[JIRA ticket
](https://dimagi-dev.atlassian.net/browse/SS-115)

This PR also integrates Knockout into the page to support dynamic UI elements and pagination, which Rynhard started prior to myself taking over this ticket. 
The work done for scheduled reports is very similar to [this PR](https://github.com/dimagi/commcare-hq/pull/30403) and models the page structure after the existing structure of the exports app. 

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

None

### QA Plan

All related QA tickets are found [here](https://dimagi-dev.atlassian.net/browse/QA-3537).

### Safety story

This PR has lots of similarities with the work done in the exports app (see linked PR above) which was tested to be safe in a previous PR. 

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/reports/static/reports/js/scheduled_reports_list.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    'jquery',
+    'knockout',
+    'underscore',
+    'hqwebapp/js/components.ko', // pagination & feedback widget
+
+], function (
+    $,
+    ko,
+    _,
+) {
```
[the remark this answers is not in the document]
Comment (biyeun) [reply corehq/apps/reports/static/reports/js/scheduled_reports_list.js:19]: the `,` after `_` will trigger JS errors. In javascript, `function` does not like that trailing comma

File: corehq/apps/reports/static/reports/js/scheduled_reports_list.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+    var scheduledReportModel = function (data, isOwner, isAdmin) {
+
+        var self = ko.mapping.fromJS(data);
+
+        self.configMany = ko.computed(function() {
```
Comment (biyeun) [inline corehq/apps/reports/static/reports/js/scheduled_reports_list.js:25]: `ko.computed` is needed when referencing observables in the computation. However, it looks like this references a static initial value. Am I reading this correctly?

Comment (biyeun) [reply corehq/apps/reports/static/reports/js/scheduled_reports_list.js:25]: the utility for `ko.computed` with observables is that it's re-computed when the value of the observable is updated. however, with a static value, the value never changes, so such a variable can just be set at initialization.

File: corehq/apps/reports/static/reports/js/scheduled_reports_list.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        };
+
+        self.is_owner = isOwner;
+        self.is_admin = isAdmin
+        self.firstConfig = ko.observable(data.configs[0]);
+
+        return self;
+    };
+
+    var scheduledReportsPanelModel = function (options) {
```
Comment (biyeun) [inline corehq/apps/reports/static/reports/js/scheduled_reports_list.js:47]: for all of the `options` and initial values passed into the object, it's good practice to assert what is required/optional, just for future use / reading of the models.

see: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/static/hqwebapp/js/assert_properties.js#L1

example: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/static/users/js/web_users.js#L25

File: corehq/apps/reports/static/reports/js/scheduled_reports_list.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    'knockout',
+    'underscore',
+    'hqwebapp/js/components.ko', // pagination & feedback widget
+
+], function (
+    $,
+    ko,
+    _,
+) {
+
```
Comment (biyeun) [inline corehq/apps/reports/static/reports/js/scheduled_reports_list.js:20]: good practice to declare `"use strict";` in your js files

Comment (minhaminha) [reply corehq/apps/reports/static/reports/js/scheduled_reports_list.js]: thanks for the tip! added!

File: corehq/apps/reports/static/reports/js/scheduled_reports_list.js (hunk 3 above)
Comment (minhaminha) [reply corehq/apps/reports/static/reports/js/scheduled_reports_list.js:67]: good call - added!

**Review by biyeun — APPROVED** [review]
Looking great! one comment on whether a variable should be computed or not, but otherwise good to go after a round of QA.

**Pull request merged**